packages/filesystem/src/browser/file-tree/file-tree-widget.ts

```diff
@@ -56,4 +63,108 @@ export class FileTreeWidget extends TreeWidget {
         return super.decorateCaption(node, VirtualRenderer.merge(icon, caption), props);
     }
 
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        if (this.model.root !== undefined) {
+            const treeNode = this.model.root;
+            this.addEventListener(this.node, 'dragenter', event => this.handleDragEnterEvent(treeNode, event));
+            this.addEventListener(this.node, 'dragover', event => this.handleDragOverEvent(treeNode, event));
+            this.addEventListener(this.node, 'dragleave', event => this.handleLeaveEvent(event));
+            this.addEventListener(this.node, 'drop', event => this.handleDropEvent(treeNode, event, false));
+        }
+    }
+
+    protected createNodeAttributes(node: ITreeNode, props: NodeProps): ElementAttrs {
+        const elementAttrs: ElementAttrs = super.createNodeAttributes(node, props);
+        const childExpansion = this.isExpandable(node) ? 0 : 1;
+
+        if (!childExpansion) {
+            // Folder, so it could be a dropzone
+            return {
+                ...elementAttrs,
+                ondrop: event => this.handleDropEvent(node, event, true),
+            };
+        }
+        return elementAttrs;
+    }
+
+    protected handleDragEnterEvent(node: ITreeNode, event: DragEvent): void {
+        event.preventDefault();
+        event.stopPropagation();
+        const target = event.target as HTMLElement;
+        if (target.parentElement) {
+            target.parentElement.classList.add(activeDropZone);
+        }
+    }
+
+    protected handleDragOverEvent(node: ITreeNode, event: DragEvent): void {
+        event.preventDefault();
+    }
+
+    protected handleLeaveEvent(event: DragEvent): void {
+        event.preventDefault();
+        event.stopPropagation();
+        const target = event.target as HTMLElement;
+        if (target.parentElement) {
+            target.parentElement.classList.remove(activeDropZone);
+        }
+    }
+
+    protected handleDropEvent(node: ITreeNode, event: DragEvent, dropAllowed: boolean): void {
```

Comment:
Why do you need the last argument? Should not be enough to check whether `node` is defined?

Reply:
Node is defined is not enough, each tree element is a node , but not necessary a folder